.github/workflows/opcode_benchmark.yml

```diff
@@ -0,0 +1,42 @@
+name: Opcode Benchmarks
+
+on:
+  push:
+   branches:
+    - main
+  pull_request:
+
+jobs:
+  test:
+    name: Benchmark library
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v4
+
+      - name: Install Nargo
+        uses: noir-lang/noirup@v0.1.4
+        with:
+          toolchain: 1.0.0-beta.3
+
+
+      - name: Build Noir benchmark programs forcing brillig backend
+        run: nargo export --force-brillig
+
+      - name: Generate gates report
+        run: ./scripts/build-brillig-gates-report.sh
+
+      - name: Compare gates reports
+        id: gates_diff
+        uses: noir-lang/noir-gates-diff@dbe920a8dcc3370af4be4f702ca9cef29317bec1
+        with:
+          report: opcode_report.json
+          summaryQuantile: 0.9 # only display the 10% most significant circuit size diffs in the summary (defaults to 20%)
```

Comment:
This needs to be marked as a brillig report

Reply:
https://github.com/noir-lang/noir/blob/0a64e8ee6497ef228a92f42af0bdb5081025bf37/.github/workflows/reports.yml#L135